api/lib/application/assessments/assessment-controller.js

```diff
@@ -90,20 +93,20 @@ module.exports = {
   async completeAssessment(request) {
     const assessmentId = parseInt(request.params.id);
 
-    await DomainTransaction.execute(async (domainTransaction) => {
-      const assessmentCompletedEvent = await usecases.completeAssessment({ domainTransaction, assessmentId });
-      const certificationScoringEvent = await events.handleCertificationScoring({ domainTransaction, assessmentCompletedEvent });
-
-      await events.handleBadgeAcquisition({ domainTransaction, assessmentCompletedEvent });
+    // TODO move Dispatcher instanciation and subscriptions elsewhere ?
+    // In DI ? In a factory ?
```

Comment:
je vote pour le mettre dans l'injection de dépendances

Reply:
C'est noté :) Je fais ça dans une prochaine PR !